res/controllers/Traktor-Kontrol-S3-hid-scripts.js

```diff
@@ -0,0 +1,2227 @@
+    this.registerInputScaler(messageLong, "[Master]", "headMix", 0x1D, 0xFFFF, this.parameterHandler);
+    this.registerInputScaler(messageLong, "[Master]", "headGain", 0x1B, 0xFFFF, this.parameterHandler);
+
+    this.hid.registerInputPacket(messageLong);
+
+    // Soft takeovers
+    for (var ch = 1; ch <= 4; ch++) {
+        var group = "[Channel" + ch + "]";
+        if (!TraktorS3.PitchSliderRelativeMode) {
+            engine.softTakeover(group, "rate", true);
+        }
+        engine.softTakeover(group, "pitch_adjust", true);
+        engine.softTakeover(group, "volume", true);
+        engine.softTakeover(group, "pregain", true);
+        engine.softTakeover("[QuickEffectRack1_" + group + "]", "super1", true);
+    }
+    for (var unit = 1; unit <= 4; unit++) {
+        group = "[EffectRack1_EffectUnit" + unit + "]";
+        var key = "mix";
+        engine.softTakeover(group, key, true);
+        for (var effect = 1; effect <= 4; effect++) {
+            group = "[EffectRack1_EffectUnit" + unit + "_Effect" + effect + "]";
+            key = "meta";
+            engine.softTakeover(group, key, true);
+            for (var param = 1; param <= 4; param++) {
+                key = "parameter" + param;
+                engine.softTakeover(group, key, true);
+            }
+        }
+    }
+
+    engine.softTakeover("[Microphone]", "volume", true);
+    engine.softTakeover("[Microphone]", "pregain", true);
+
+    engine.softTakeover("[EqualizerRack1_[Channel1]_Effect1]", "parameter1", true);
+    engine.softTakeover("[EqualizerRack1_[Channel1]_Effect1]", "parameter2", true);
+    engine.softTakeover("[EqualizerRack1_[Channel1]_Effect1]", "parameter3", true);
+    engine.softTakeover("[EqualizerRack1_[Channel2]_Effect1]", "parameter1", true);
+    engine.softTakeover("[EqualizerRack1_[Channel2]_Effect1]", "parameter2", true);
+    engine.softTakeover("[EqualizerRack1_[Channel2]_Effect1]", "parameter3", true);
+    engine.softTakeover("[EqualizerRack1_[Channel3]_Effect1]", "parameter1", true);
+    engine.softTakeover("[EqualizerRack1_[Channel3]_Effect1]", "parameter2", true);
+    engine.softTakeover("[EqualizerRack1_[Channel3]_Effect1]", "parameter3", true);
+    engine.softTakeover("[EqualizerRack1_[Channel4]_Effect1]", "parameter1", true);
+    engine.softTakeover("[EqualizerRack1_[Channel4]_Effect1]", "parameter2", true);
+    engine.softTakeover("[EqualizerRack1_[Channel4]_Effect1]", "parameter3", true);
+
+    // engine.softTakeover("[Master]", "crossfader", true);
+    engine.softTakeover("[Master]", "gain", true);
+    // engine.softTakeover("[Master]", "headMix", true);
+    // engine.softTakeover("[Master]", "headGain", true);
+
+    for (var i = 1; i <= 16; ++i) {
+        engine.softTakeover("[Sampler" + i + "]", "pregain", true);
+    }
+
+    for (ch in this.Channels) {
+        var chanob = this.Channels[ch];
+        engine.connectControl(ch, "playposition",
```

Comment:
```suggestion
        engine.makeConnection(ch, "playposition",
```

`engine.connectControl` is deprecated